### ISSUE TITLE
StableDiffusionPipeline callback problem

### ISSUE THREAD

**Issue by hckj588ku** [huggingface/diffusers#3553]
### Describe the bug

I want to display the results of the image generation process in diffusers, so I called the callback function. However, when I convert the received tensor variable to an image in the callback, I found that the image has a size of only 64x64 pixels, and it looks very strange. Is there a problem with the way I'm converting it?

### Reproduction

`def sample_callback(step: int, timestep: int, latents: torch.FloatTensor):
    outputRs = latents.permute(0, 2, 3, 1)
    k = outputRs.cpu().detach().numpy()
    size = latents.shape[0]
    for i in range(size):
        res = k[i] 
        image = Image.fromarray(np.uint8(res)).convert('RGB')
        # image.show()
        savepath = str(i) + '_' + str(step) + '_r.jpg'
        image.save(savepath)
    print(f'step: {step}, timestep: {timestep}, latents: {size}')
`

### System Info

python3.10  system:linux

**Comment by hckj588ku** [comment]
`def sample_callback(step: int, timestep: int, latents: torch.FloatTensor):

    outputRs = latents.permute(0, 2, 3, 1)
    k = outputRs.cpu().detach().numpy()
    size = latents.shape[0]
    for i in range(size):
        res = k[i]
        image = Image.fromarray(np.uint8(res)).convert('RGB')
        # image.show()
        savepath = str(i) + '_' + str(step) + '_r.jpg'
        image.save(savepath)
    print(f'step: {step}, timestep: {timestep}, latents: {size}')`

**Comment by hckj588ku** [comment]
`  
def sample_callback(step: int, timestep: int, latents: torch.FloatTensor):

    outputRs = latents.permute(0, 2, 3, 1)
    k = outputRs.cpu().detach().numpy()
    size = latents.shape[0]
    for i in range(size):
        res = k[i]
        image = Image.fromarray(np.uint8(res)).convert('RGB')
        # image.show()
        savepath = str(i) + '_' + str(step) + '_r.jpg'
        image.save(savepath)
    print(f'step: {step}, timestep: {timestep}, latents: {size}')`

**Comment by patrickvonplaten** [comment]
Hey @hckj588ku,

This is because the stable diffusion model works in latent space. We need to decode it to pixel space (512x512) before.
